File: server/app/routes/booking_routes.py
from flask import Blueprint
from app.controller.booking_controller import create_booking_controller, check_availability_controller
from app.middleware.auth_middleware import auth_required

booking_bp = Blueprint("booking", __name__)

booking_bp.route("/space/<int:space_id>", methods=["GET"])(check_availability_controller)
booking_bp.route("/", methods=["POST"])(auth_required(create_booking_controller))

File: server/app/routes/booking_routes.js
import express from "express"
import { createBooking, checkAvailability } from "../controllers/booking_controller.js"
import authMiddleware from "../middleware/auth_middleware.js"

const router = express.Router()

router.post("/", authMiddleware, createBooking)
router.get("/space/:spaceId", checkAvailability)

export default router
